app/frontend/wallet/wallet.ts

```diff
@@ -0,0 +1,95 @@
+import BlockchainExplorer from './blockchain-explorer'
+import {AccountManager} from './account-manager'
+
+const Wallet = ({config, cryptoProvider}) => {
+  const blockchainExplorer = BlockchainExplorer(config)
+
+  const accountManager = AccountManager({config, cryptoProvider, blockchainExplorer})
+
+  function isHwWallet() {
+    return cryptoProvider.isHwWallet()
+  }
+
+  function getWalletName() {
+    return cryptoProvider.getWalletName()
+  }
+
+  function submitTx(signedTx): Promise<any> {
+    const params = {
+      walletType: getWalletName(),
+      // TODO: stakeKey
+    }
+    const {txBody, txHash} = signedTx
+    return blockchainExplorer.submitTxRaw(txHash, txBody, params)
+  }
+
+  function getWalletSecretDef() {
+    return {
+      rootSecret: cryptoProvider.getWalletSecret(),
+      derivationScheme: cryptoProvider.getDerivationScheme(),
+    }
+  }
+
+  async function fetchTxInfo(txHash) {
+    return await blockchainExplorer.fetchTxInfo(txHash)
+  }
+
+  function checkCryptoProviderVersion(type: string) {
+    try {
+      cryptoProvider.checkVersion(type)
+    } catch (e) {
+      return {code: e.name, message: e.message}
+    }
+    return null
+  }
+
+  const getWalletInfo = (accountsInfo) => {
+    const totalWalletBalance = accountsInfo.reduce(
+      (a, {shelleyBalances}) =>
+        shelleyBalances.stakingBalance + shelleyBalances.nonStakingBalance + a,
+      0
+    )
+    const totalRewardsBalance = accountsInfo.reduce(
+      (a, {shelleyBalances}) => shelleyBalances.rewardsAccountBalance + a,
+      0
+    )
+    const shouldShowSaturatedBanner = accountsInfo.some(
```

Comment:
I find it a bit odd to have a flag that represents an apparently higher-level concern (telling directly what should be shown in the UI) in this low-level wallet class. Given that it's been like this before this PR it's probably sensible to keep it, but probably deserves to be extracted together with the logic to determine what pool recommendation to show into actions

Reply:
You are right, I ll move the function to `actions.ts`.